src/zope/i18n/tests/test_formats.py

```diff
@@ -365,11 +387,56 @@ def testParseUnusualFormats(self):
                               'dddd. MMM yyyy hhhh:mm a'),
             datetime.datetime(2003, 1, 1, 00, 00, 00, 00))
 
+    def testParseNotObject(self):
+        self.assertEqual(
+            ('2017', '01', '01'),
+            self.format.parse('2017-01-01', 'yyyy-MM-dd', asObject=False))
+
+    def testParseTwoDigitYearIs20thCentury(self):
+        self.assertEqual(
+            datetime.date(1952, 1, 1),
+            self.format.parse('52-01-01', 'yy-MM-dd'))
+
+        # 30 is the cut off
+        self.assertEqual(
+            datetime.date(1931, 1, 1),
+            self.format.parse('31-01-01', 'yy-MM-dd'))
+
+        self.assertEqual(
+            datetime.date(2030, 1, 1),
+            self.format.parse('30-01-01', 'yy-MM-dd'))
+
+    def testParseAMPMMissing(self):
+        with self.assertRaisesRegex(
+                DateTimeParseError,
+                'Cannot handle 12-hour format without am/pm marker.'):
+            self.format.parse('02.01.03 09:48', 'dd.MM.yy hh:mm')
+
+    def testParseBadTimezone(self):
+        # Produces an object without pytz info
+        self.assertEqual(
+            datetime.time(21, 48, 1),
+            self.format.parse(
+                '21:48:01 Bad/Timezone',
+                'HH:mm:ss zzzz'))
+
+    def testParsePyTzTimezone(self):
+        # XXX: Bug: This raises
+        # TypeError:  unsupported operand type(s) for +: 'datetime.time' and 'datetime.timedelta'
+        # at pytz/tzinfo.py:309
+        # (PyPy and CPython produce slightly different error messages;
+        # PyPy doesn't include the modulename)
+        tzinfo = pytz.timezone("US/Central")
+        with self.assertRaisesRegex(TypeError, 'timedelta'):
+            self.format.parse(
+                '21:48:01 US/Central',
+                'HH:mm:ss zzzz')
```

Comment:
I'm tempted to just fix it.  (But you're right, the fix should be in a separate PR.)

Reply:
It's not entirely clear to me how to fix it. Just don't call `tzinfo.localize` on a time?

Reply:
I was thinking `tzinfo.localize(datetime.datetime.combine(datetime.date.today(), the_parsed_time)).timetz()`.